### PR TITLE
chore: expose docs files on the website

### PR DESCRIPTION
**Pull request by Simek**
## Summary

Currently the Hermes has it's own, basic website setup and a set of docs files avaible in the repository. I don't know the history, but it looks like, from the fromtmatter headers, in most of the docs files that in some period of time the markdown docs files were used previously as Docusaurus pages.

This PR exposes the documentation files currently located in `/doc/` folder on the Hermes website. The addition includes basic sidebar grouping and changes to the Footer content (+ FB OSS logo in repository). Since I'm not that familiar with the Hermes structure or communication goals you might want to edit sidebar and footer content to better suit it to your needs.

I have also added small tweaks to the `prism` - code highlight library used in Docusaurus. Shell script is set as default language for prism blocks and light theme has been added.

Besides the changes above, I have bumped the Docusuaurs version to latest release. Currently used alpha was released more than 7 months ago.

## Test Plan

Build and run of Hermes website locally.

#### Preview

Screenshot 2020-12-07 133712

Screenshot 2020-12-07 133221

#### Footer Preview

Screenshot 2020-12-07 133551

### CODE REVIEW

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -59,16 +77,29 @@ module.exports = {
...
-        src: 'https://docusaurus.io/img/oss_logo.png',
+        src: 'img/oss_logo.png',
         href: 'https://opensource.facebook.com',
       },
       copyright: `Copyright © ${new Date().getFullYear()} Facebook, Inc. Built with Docusaurus.`,
     },
+    prism: {
+      defaultLanguage: 'shell',
+      theme: require('prism-react-renderer/themes/github'),
+      darkTheme: require('prism-react-renderer/themes/dracula'),
```
Comment (Huxpro) [inline website/docusaurus.config.js:88]: I'm wondering are these two themes used by React Native and React website? If not, can we use the same ones?

File: website/sidebars.json [hunk 2]
```diff
@@ -1 +1,22 @@
-{}
+{
```
Comment (Huxpro) [inline website/sidebars.json:1]: I need some time to think about how to better categorize those documents.

**Review by Huxpro — COMMENTED** [review]
LGTM and thanks for the work!
Left a question but it's not merge-blocking. I'll import and merge late this week.

Btw, do you happen to be able to help with this website issue as well? https://github.com/facebook/hermes/issues/432

File: website/docusaurus.config.js (hunk 1 above)
Comment (Simek) [reply website/docusaurus.config.js:88]: On the React Native website we use custom prism theme, which always render code block on dark background, no matter of theme selected by user. If it's not an issue for you, I can import the theme in.

**Comment by Simek** [comment]
#### Preview after changes (custom prism theme, updated sidebar)
Screenshot 2020-12-10 125030

**Comment by Huxpro** [comment]
@Simek I'm seeing this error in a production build:

```
hermes/website git/expose-docs-on-website*  16s
λ yarn build
yarn run v1.22.4
$ docusaurus build
Creating an optimized production build...

● Client █████████████████████████ chunk asset optimization (92%) OptimizeCssAssetsWebpackPlugin


● Server █████████████████████████ building (70%) 955/955 modules 0 active


/Users/jsx/github/hermes/website/node_modules/clean-css/lib/optimizer/level-2/properties/populate-components.js:30
          throw e;
          ^

TypeError: Cannot read property '0' of undefined
    at Object.font [as breakUp] (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/optimizer/level-2/break-up.js:303:91)
    at populateComponents (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/optimizer/level-2/properties/populate-components.js:17:42)
    at /Users/jsx/github/hermes/website/node_modules/clean-css/lib/optimizer/level-2/remove-unused-at-rules.js:141:9
    at markUsedAtRules (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/optimizer/level-2/remove-unused-at-rules.js:63:9)
    at removeUnusedAtRule (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/optimizer/level-2/remove-unused-at-rules.js:43:3)
    at removeUnusedAtRules (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/optimizer/level-2/remove-unused-at-rules.js:22:3)
    at level2Optimize (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/optimizer/level-2/optimize.js:117:5)
    at optimize (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/clean.js:147:5)
    at /Users/jsx/github/hermes/website/node_modules/clean-css/lib/clean.js:120:29
    at /Users/jsx/github/hermes/website/node_modules/clean-css/lib/reader/read-sources.js:26:64
    at loadOriginalSources (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/reader/load-original-sources.js:26:5)
    at /Users/jsx/github/hermes/website/node_modules/clean-css/lib/reader/read-sources.js:26:14
    at applySourceMaps (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/reader/apply-source-maps.js:34:5)
    at Object.callback (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/reader/read-sources.js:25:12)
    at doInlineImports (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/reader/read-sources.js:200:25)
    at inline (/Users/jsx/github/hermes/website/node_modules/clean-css/lib/reader/read-sources.js:178:10)
error Command failed with exit code 1.
```

**Comment by Huxpro** [comment]
Huh! Found at https://github.com/facebook/docusaurus/issues/3836.

Seems like it's fixed on master and the temp workaround is `yarn build --no-minify`

**Comment by Simek** [comment]
> Could this be a bug of the upgraded Docusaurus?

@Huxpro Yes, it's issue related to the CSS minification introduced in `alpha.68` release:
* https://github.com/facebook/docusaurus/releases/tag/v2.0.0-alpha.68

Happy to see you have implemented a workaround :) I would look through the CSS to find the culprit and fix it, if possible.

---

Refs: #432: Nice to see that Sebastien left a comment which have helped you fix the issue. 

I was also looking into that issue and after analyzing the `normalize.css` code I think that the tweaks in there should not be required for plain Docususaurs V2 setup. If Playground need those tweaks (need to verify that locally), the styles can be added as a custom stylesheet, which do not seems that bad, since `normalize.css`  includes only several rules.

**Pull request closed without merging**